protocol-units/bridge/chains/ethereum/src/client.rs

```diff
@@ -223,18 +250,44 @@ impl BridgeContractInitiator for EthClient {
 		time_lock: TimeLock,
 		amount: Amount, // the ETH amount
 	) -> BridgeContractInitiatorResult<()> {
+		if amount.weth() > 0 {
+			let deposit_weth_signer = self.get_signer_address();
+			println!("deposit_weth_signer: {:?}", deposit_weth_signer);
+			let contract = self.weth_contract().expect("WETH contract not set");
+			let call = contract.deposit().value(U256::from(amount.weth()));
+			send_transaction(call, &send_transaction_rules(), RETRIES, GAS_LIMIT)
+				.await
+				.expect("Failed to deposit eth to weth contract");
+
+			let approve_call: alloy::contract::CallBuilder<_, &_, _> = contract
+				.approve(self.initiator_contract_address()?, U256::from(amount.weth()));
+			let WETH9::balanceOfReturn { _0: balance } = contract
+				.balanceOf(deposit_weth_signer)
+				.call()
+				.await
+				.expect("Failed to get balance");
+			let signer = self.get_signer_address();
+			println!("caller: {}", signer);
+			println!("balance: {}", balance);
+
+			send_transaction(approve_call, &send_transaction_rules(), RETRIES, GAS_LIMIT)
+				.await
+				.expect("Failed to approve");
+		}
 		let contract =
 			AtomicBridgeInitiator::new(self.initiator_contract_address()?, &self.rpc_provider);
 		let recipient_bytes: [u8; 32] =
 			recipient_address.0.try_into().expect("Recipient address must be 32 bytes");
 		let call = contract
 			.initiateBridgeTransfer(
-				U256::from(0), // For now a 0 WETH amount
+				U256::from(amount.weth()),
 				FixedBytes(recipient_bytes),
 				FixedBytes(hash_lock.0),
 				U256::from(time_lock.0),
 			)
-			.value(U256::from(amount.0));
+			.value(U256::from(amount.eth())).from(_initiator_address.0.0);
```

Comment:
We should remove the underscore from `_initiator_address` as its now being used.